File: wiz-shop-java/src/main/java/com/example/wizshopjava/api/member/dto/MemberRegisterRequest.java
package com.example.wizshopjava.api.member.dto;

import com.example.wizshopjava.domain.member.entity.Member;

public record MemberRegisterRequest(
	String email,
	String password,
	String nickname
) {

	public Member toEntity() {
		return Member.builder()
			.email(email)
			.password(password)
			.nickname(nickname)
			.build();
	}
}

File: wiz-shop/src/main/kotlin/com/example/wizshop/api/member/dto/MemberRegisterRequest.kt
package com.example.wizshop.api.member.dto

import com.example.wizshop.domain.member.entity.Member

data class MemberRegisterRequest(
    val email: String,
    val password: String,
    val nickname: String
) {

    fun toEntity() = Member(
        email = this.email,
        password = this.password,
        nickname = this.nickname
    )
}